src/BloomTests/Book/HtmlDomTests.cs

```diff
@@ -157,5 +158,54 @@ public void SortStyleSheetLinks_LeavesOverridesAtEndAndSpecialFilesInMiddle()
 			AssertThatXmlIn.Dom(dom).HasSpecifiedNumberOfMatchesForXpath("//head/link[10][@href='../customCollectionStyles.css']", 1);
 			AssertThatXmlIn.Dom(dom).HasSpecifiedNumberOfMatchesForXpath("//head/link[11][@href='customBookStyles.css']", 1);
 		}
+
+
+		[Test]
+		public void MergeClassesIntoNewPage_BothEmtpy()
+		{
+			AssertHasClases("", MergeClasses("","", new[] { "dropMe" }));
+		}
+		[Test]
+		public void MergeClassesIntoNewPage_TargetEmpty()
+		{
+			AssertHasClases("one two", MergeClasses("one two", "", new[] {"dropMe"}));
+		}
+		[Test]
+		public void MergeClassesIntoNewPage_SourceEmpty()
+		{
+			AssertHasClases("one two", MergeClasses("", "one two", new[] { "dropMe" }));
+		}
+		[Test]
+		public void MergeClassesIntoNewPage_SourceAllDroppable()
+		{
+			AssertHasClases("one two", MergeClasses("dropMe dropMe dropMe ", "one two", new[] { "dropMe" }));
+		}
+		[Test]
+		public void MergeClassesIntoNewPage_MergesAndDropsItemsInDropList()
+		{
+			AssertHasClases("one two three", MergeClasses("one drop two delete", "three", new[] { "drop","delete" }));
+		}
+
+		private void AssertHasClases(string expectedString, string actualString)
+		{
```

Comment:
The method AssertHasClases should be AssertHasClasses, but I'm willing to let it pass.